src/plugins/PasteHandler.ts

```diff
@@ -34,11 +34,19 @@ export default class PasteHandler extends Extension {
     return [
       new Plugin({
         props: {
-          transformPastedHTML: (html: string) => {
-            // TODO: this is a temp fix for pasting headers..
-            //  uncomment to see the error :) this is to be fixed ASAP
-            return replaceHeaderByStrong(html);
-          },
+          /**
+           * I'm not sure the use case for this previously
+           * from what I understand this pasted any header tags (copied from within the editor or outside)
+           * to just be <strong>. Without this line of code, header pasted copied from the editor work
+           * HOWEVER, headers copied from other sites would be pasted as regular text
```

Comment:
so previously, before this change was made, copying notes from editor to editor worked fine, including pasting headers. However, when copying content with headers from another website, it worked, until we refresh the page! when refreshing, I think an error was thrown - I will test quickly though, and attach a recording of the broken behavior!

Reply:
ok got it got it -> i knew that piece of code was there for a reason but couldn't find the use case for it haha. The screen recordings helped a lot got it thanks so much - I'll keep working on this

Reply:
Ok, I think I found the issue but idrk what the best method to solve it is.

Basically, the reason why heading pastes aren't working from certain websites is that our specific editor schema is expecting an inner ```<span class='heading-content'>Inner Content</span> ``` for each heading element, but some websites just directly render text content as innerHTML. That's why we are getting the ```firstChild``` bug, because we expect a child to exist within heading elements.

So I think the best move is to parse the HTML and append ```span``` tags with each respective heading tag IF they arent there already. Idk how to do this lol.

Reply:
oh, great finding! i am not too familiar with the dom parser API, i can't think of a good way to add spans if they're missing :) though, do you think we can just do 🙂 

```js
// contentElement: ".heading-content"
```